src/content/docs/en/guides/typescript.mdx

```diff
@@ -235,6 +235,26 @@ type MyAttributes = astroHTML.JSX.ImgHTMLAttributes;
 ```
 :::
 
+### Polymorphic type
+
+<Since v="2.5.0" />
+
+Astro includes a helper to make it easier to build components that can render as different HTML elements with full type safety. This is useful for components like `<Link>` that can render as either `<a>` or `<button>` depending on the props passed to it.
+
+The example below implements a fully typed, polymorphic component that can render as any HTML element. The [`HTMLTag`](#built-in-html-attributes) type is used to ensure that the `as` prop is a valid HTML element.
```

Comment:
```suggestion
The example below implements a fully-typed, polymorphic component that can render as any HTML element. The [`HTMLTag`](#built-in-html-attributes) type is used to ensure that the `as` prop is a valid HTML element.
```
Lovely! This is fantastic! (just added a hypen).

I consider this "awkward green smiley face" ready to go for release!